apps/precursor-chemicals/behaviours/filter-chemicals.js

```diff
@@ -0,0 +1,12 @@
+module.exports = superclass => class extends superclass {
+  configure(req, res, next) {
+    const chosenCategory = req.sessionModel.get('substance-category');
+    let options = req.form.options.fields['which-chemical'].options;
+    if (chosenCategory !== 'unknown') {
+      options = options.filter(option => !option.category || option.category === chosenCategory);
+    }
+    req.form.options.fields['which-chemical'].options = options;
```

Comment:
this is redundant step, no changes required to `req.form.options.fields['which-chemical'].options`